SingularityUI/app/components/common/table/UITable.jsx

```diff
@@ -79,23 +75,24 @@ class UITable extends Component {
         chunkNum: 1,
         rowChunkSize: this.props.rowChunkSize,
         lastPage: false,
-        pastEnd: false
       });
     };
   }
 
   fetchDataFromApi(chunkNum, rowChunkSize, updateStateAfterFetching = false, sortBy = this.state.sortBy) {
-    let lastPage = this.state.lastPage;
-    if (chunkNum < this.state.chunkNum) {
-      lastPage = false;
-    }
-    if (!updateStateAfterFetching) {
-      this.setState({chunkNum, rowChunkSize, sortBy, lastPage});
-    }
-    return this.props.fetchDataFromApi(chunkNum, rowChunkSize, sortBy).then(() => {
-      if (updateStateAfterFetching) {
-        this.setState({chunkNum, rowChunkSize, sortBy, lastPage});
+    const lookaheadNum = (chunkNum * rowChunkSize) + 1;
+    this.setState({ lookahead: true });
+    return this.props.fetchDataFromApi(lookaheadNum, 1).then((lookaheadResponse) => {
```

Comment:
Is this something that we always want to do for UITable or should this be configurable. You might also want to consider doing both of these requests in parallel and using `Promise.all`. That might make it unnecessary to separately keep track of the lookahead request state.

Reply:
I think it makes sense in the current UITable design since it only applies to tables that have the `fetchDataFromApi` prop set. Do you foresee tables that have this prop set but won't have a lookahead call?

I had considered running the requests in parallel, but I thought it might be more complicated because the order of completion is undefined. Since the `data` prop of the UI table is supplied by the parent component and _that_ is supplied by the `requestApi` object.